rust/integration-tests/src/diff_tests/all_tests.rs

```diff
@@ -0,0 +1,181 @@
+
+        run_processor_tests(
+            processor_configs,
+            &test_context,
+            get_expected_imported_mainnet_txns,
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn test_all_generated_txns_schema_output_for_all_processors() {
+        let processor_configs = get_processor_configs();
+        let test_context = TestContext::new(ALL_GENERATED_TXNS).await.unwrap();
+
+        run_processor_tests(
+            processor_configs,
+            &test_context,
+            get_expected_generated_txns,
+        )
+        .await;
+    }
+
+    // Helper function to reduce duplicate code for running tests on all processors
+    async fn run_processor_tests<F>(
+        processor_configs: Vec<TestProcessorConfig>,
+        test_context: &TestContext,
+        get_expected_json_path_fn: F,
+    ) where
+        F: Fn(&str, &str) -> String + Send + Sync + 'static + std::marker::Copy,
+    {
+        let processor_map = get_processor_map();
+        for processor_config in processor_configs {
+            let processor_name = processor_config.config.name();
+            let test_type = TestType::Diff(DiffTest);
+
+            if let Some(test_helper) = processor_map.get(processor_name) {
+                let test_helper = Arc::clone(test_helper);
+
+                test_context
+                    .run(
+                        processor_config,
+                        test_type,
+                        move |conn: &mut PgConnection, txn_version: &str| {
+                            let mut json_data = match test_helper.load_data(conn, txn_version) {
+                                Ok(data) => data,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Failed to load data for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            let expected_json_path = get_expected_json_path_fn(processor_name, txn_version);
+                            let expected_json = match read_and_parse_json(&expected_json_path) {
+                                Ok(json) => json,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Error handling JSON for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            remove_inserted_at(&mut json_data);
+                            // Remove the Err handling around assert_json_eq!, as it doesn't return a Result type
```

Comment:
can we add a todo here about the json diff; this can be used for testing transaction diff but also diffing here.